### PR TITLE
TableFeature javadoc and more tests

### LINKED ISSUE #453: TableFeature needs javadoc and more tests
The TableFeature class has almost no documentation and barely any tests.

those two are completely unused and untested.
getAllValues
getValuesTo(int columnPosition) 

Both doc and testing needs to improve for this class.
Came up in review of #449

### PR DESCRIPTION
**Pull request by magicDGS**
Closes #453

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/utils/codecs/table/TableFeature.java [hunk 1]
```diff
@@ -21,21 +23,20 @@
...
+     */
+    public TableFeature(final Locatable position, final List<String> values, final List<String> keys) {
         this.values = values;
         this.keys = keys;
         this.position = position;
     }
 
     @Override
-    @SuppressWarnings("deprecation")
-    //Suppressing because we have to implement this
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/utils/codecs/table/TableFeature.java]: Good riddance to this method

File: src/test/java/org/broadinstitute/hellbender/utils/codecs/table/TableFeatureUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+
+
+/**
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public class TableFeatureUnitTest {
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/utils/codecs/table/TableFeatureUnitTest.java]: This should extend `BaseTest`

File: src/main/java/org/broadinstitute/hellbender/utils/codecs/table/TableFeature.java [hunk 3]
```diff
@@ -50,37 +51,48 @@ public int getEnd() {
...
         return this.position;
     }
 
+    /** Gets all the values in order.  */
     public List<String> getAllValues() {
-        return getValuesTo(values.size());
+        return getValuesTo(columnCount());
     }
 
+    /** Gets all the values from the first column to the last one specified (exclusive). */
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/utils/codecs/table/TableFeature.java]: Mention that the indexing is 0-based

File: src/test/java/org/broadinstitute/hellbender/utils/codecs/table/TableFeatureUnitTest.java (hunk 2 above)
Comment (magicDGS) [inline src/test/java/org/broadinstitute/hellbender/utils/codecs/table/TableFeatureUnitTest.java]: True, I overseen it! Done!

File: src/main/java/org/broadinstitute/hellbender/utils/codecs/table/TableFeature.java (hunk 3 above)
Comment (magicDGS) [inline src/main/java/org/broadinstitute/hellbender/utils/codecs/table/TableFeature.java]: Done.

**Comment by magicDGS** [comment]
Back to you @droazen

**Comment by magicDGS** [comment]
@droazen, I think that the tests are failing here because a cromwell issue...

**Comment by lbergelson** [comment]
@magicDGS Thank you.

**Comment by magicDGS** [comment]
Thanks!

**Pull request merged**